src/cargo/util/toml/mod.rs

```diff
@@ -2292,6 +2292,9 @@ supported tools: {}",
                     }
                 }
             }
+            if config.level().is_none() && config.config().map(|c| c.is_empty()).unwrap_or(true) {
+                anyhow::bail!("`lints.{tool}.{name}` missing field `level`");
+            }
```

Comment:
This is the big question for me: what should be allows

Today
```toml
lint_name = {}  # disallowed
lint_name = { config_name = "config_value" }  # disallowed
lint_name = { priority = 1 }  # disallowed
lint_name = { priority = 1, config_name = "config_value" }  # disallowed
lint_name = { level = "warn" }  # allowed
lint_name = { level = "warn", priority = 1 }  # allowed
```

With this PR
```toml
lint_name = {}  # disallowed
lint_name = { config_name = "config_value" }  # allowed
lint_name = { priority = 1 }  # disallowed
lint_name = { priority = 1, config_name = "config_value" }  # allowed
lint_name = { level = "warn" }  # allowed
lint_name = { level = "warn", priority = 1 }  # allowed
```

This allows `priority` without `level`.  Should we allow that?

This requires `level` or config, should we bother with an empty table error?

I almost lean towards either keeping the requirement on `level` or dropping all requirements (though `priority` alone feels weird).

Thoughts?

Reply:
> This allows `priority` without `level`. Should we allow that?

I don't think so, if `priority` is set then `level` must also be set, that's a oversight on my part. Will fix it.

> This requires `level` or `config`, should we bother with an empty table error?

My stance is that it is useless and weird to have empty table.

Reply:
Fixed the `priority` field alone without the `level` field. It now bail out.